website_portal/controllers/main.py

```diff
@@ -98,21 +98,21 @@ def details_form_validate(self, data):
 
 class AuthSignup(AuthSignupHome):
 
-    @http.route(website=True, auth="public")
-    def web_login(self, *args, **kw):
-        res = super(AuthSignup, self).web_login(*args, **kw)
+    def redirect(self, res, **kw):
         if request.session.uid:
             user = request.env['res.users'].browse(request.session.uid)
-            if not user.has_group('base.group_user'):
-                return http.redirect_with_hash('/')
+            redirect = kw.get('redirect', '')
+            if not user.has_group('base.group_user') and\
+                    (redirect == '' or redirect[-5:] == '/web?'):
```

Comment:
No, i only want five last positions. Only redirect to '/' in case that url is 'base.url/web?'.